### PR TITLE
858: infinite retention policy

### PR DESCRIPTION
**Pull request by ZakharE**
Attempt to solve #858. 
I walked through usages of `RetentionPolicy` and added conditions where it was necessary according to the comments. 
I also had an idea: add method to `RetentionPolicy` type, since there is pattern `rp != LimitsPolicy && rp != InifinitePolicy`. But i am not sure how to name it correctly

### CODE REVIEW

File: server/memphis_handlers_stations.go [hunk 1]
```diff
@@ -685,17 +686,18 @@ func (sh StationsHandler) CreateStation(c *gin.Context) {
...
 			serv.Warnf("CreateStation: Station " + body.Name + ": " + err.Error())
 			c.AbortWithStatusJSON(SHOWABLE_ERROR_STATUS_CODE, gin.H{"message": err.Error()})
 			return
 		}
-	} else {
-		retentionType = "message_age_sec"
-		body.RetentionValue = 604800 // 1 week
+	}
+
+	if retentionType == "infinite" {
```
Comment (ZakharE) [inline server/memphis_handlers_stations.go:699]: I am not sure about this hack. I've added it to ensure that i will not break any logic that rely to this field 😅

**Comment by idanasulinmemphis** [comment]
Hi @ZakharE and thanks for this amazing PR.
I think that you made it a bit more complex than what it was supposed to be.
I will describe the flow and please fix it accordingly.

1. user should be able to create a Station with infinite retention (from the UI and the SDKs)
2. when creating a new station (broker side), you will find that we are creating a stream - you have to set its limits to -1 in order that no limits will apply on it.
3. add a new enum type to the stations table so the retention type could be infinite - regarding this one please make sure that it is backwards compatible, that means that you need to think about users who already have this table exist in their environment and then you will need to update it with some alter query I guess.

Thanks again, and waiting for your reply

**Comment by ZakharE** [comment]
Thank you for review! 
I'll take a look and will try to implement it this week

**Comment by SvetaMemphis** [comment]
Hi @ZakharE, how is it going?

**Comment by ZakharE** [comment]
Hi! 
Sorry. I can't no longer work on this issue. I'm closing the PR

**Pull request closed without merging**